Resources/public/js/views/services/ez-roleserversideviewservice.js

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-roleserversideviewservice', function (Y) {
+    "use strict";
+    /**
+     * Provides the role server side view service class
+     *
+     * @method ez-roleserversideviewservice
+     */
+    Y.namespace('eZ');
+
+    /**
+     * The Role Server Side View Service class.
+     *
+     * @namespace eZ
+     * @class RoleServerSideViewService
+     * @constructor
+     * @extends eZ.ServerSideViewService
+     */
+    Y.eZ.RoleServerSideViewService = Y.Base.create('roleServerSideViewService', Y.eZ.ServerSideViewService, [], {
+        initializer: function () {
+            this.on('*:contentDiscover', function (e) {
+                e.config.contentDiscoveredHandler = Y.bind(this._assignRole, this);
+            });
+        },
+
+        /**
+         * Assign the role to the contents after the user has chosen them in
+         * the universal discovery widget
+         *
+         * @method _assignRole
+         * @protected
+         * @param {EventFacade} e
+         */
+        _assignRole: function (e) {
+            var data = e.target.get('data'),
+                tasks = new Y.Parallel(),
+                userService = this.get('capi').getUserService();
+
+            this._assignRoleNotificationStarted(
+                data.roleId, data.roleName, e.selection
+            );
+
+            Y.Array.each(e.selection, function (struct) {
+                var role = userService.loadRole(data.roleId);
+                var roleAssign = userService.newRoleAssignInputStruct(role); // TODO assign with limitation
+                var userGroupId = struct.content.get('id');
+
+                if (userGroupId.indexOf('/user/groups/') > 0) {
+                    userService.assignRoleToUserGroup(
+                        userGroupId, roleAssign, tasks.add()
```

Comment:
@mhyndle Tried it, but I get the same error as before, "TypeError: callback is not a function CAPI.js:1389:12"
Found it now in my local CAPI.js, it corresponds to this:
https://github.com/ezsystems/ez-js-rest-client/blob/master/dist/CAPI.js#L1389
Does this tell you anything?